### PR TITLE
fix: maintain the order of patches to apply

### PR DESCRIPTION
**Pull request by matino**
That was easier then I thought :)

Closes https://github.com/nikolaydubina/go-instrument/issues/22

### CODE REVIEW

File: main_test.go [hunk 1]
```diff
@@ -52,6 +52,16 @@ func TestApp(t *testing.T) {
 		assertEqFile(t, "./internal/testdata/instrumented/basic_include_only.go.exp", f)
 	})
 
+	t.Run("when complex, then ok", func(t *testing.T) {
+		f := copyFile(t, "./internal/testdata/complex.go")
```
Comment (matino) [inline main_test.go:56]: not sure about the `complex.go` file name, maybe it should be a part of `basic.go`?

Comment (nikolaydubina) [reply main_test.go:56]: yeah, this should be in `basic.go`

**Comment by nikolaydubina** [comment]
wow, impressive how quickly you found it!

patches is a bit complex code. likely changing order is ok, but let me check tomorrow

File: processor/processor.go [hunk 2]
```diff
@@ -185,6 +186,12 @@ func (p *Processor) Process(fset *token.FileSet, file *ast.File) error {
 	})
 
 	if len(patches) > 0 {
+		// Patches must be applied in the ascending order, otherwise the
+		// modified source file will become corrupted.
+		sort.Slice(patches, func(i, j int) bool {
+			return patches[i].pos < patches[j].pos
+		})
```
Comment (nikolaydubina) [inline processor/processor.go:193]:
```suggestion
		sort.Slice(patches, func(i, j int) bool {return patches[i].pos < patches[j].pos})
```

File: processor/processor.go [hunk 3]
```diff
@@ -185,6 +186,12 @@ func (p *Processor) Process(fset *token.FileSet, file *ast.File) error {
 	})
 
 	if len(patches) > 0 {
+		// Patches must be applied in the ascending order, otherwise the
+		// modified source file will become corrupted.
```
Comment (nikolaydubina) [inline processor/processor.go:190]: yeah, but why it is corrupted...🤷🏻‍♂️ this code is complex. let me review tomorrow

File: internal/testdata/instrumented/basic.go.exp [hunk 4]
```diff
@@ -219,3 +219,35 @@ func Closure(ctx context.Context) (int, error) {
...
+	ctx, span := otel.Tracer("app").Start(ctx, "FunctionWithAnonymousFunc")
+	defer span.End()
+
+	if err := cmd.Validate(); err != nil {
+		return err
+	}
+
+	if err := Exec(ctx, func(ctx context.Context) error {
+		ctx, span := otel.Tracer("app").Start(ctx, "anonymous")
+		defer span.End()
```
Comment (nikolaydubina) [inline internal/testdata/instrumented/basic.go.exp:239]: nice. before this was failing

File: processor/processor.go (hunk 3 above)
Comment (matino) [reply processor/processor.go:190]: > guess has something to do with recalculation of offsets when patches applied

exactly this

**Review by nikolaydubina — APPROVED** [review]
looks good. lets merge it

**Pull request merged**